TypeCobol/Compiler/Types/ArrayType.cs

```diff
@@ -23,9 +28,24 @@ public long MinOccur
         }
 
         /// <summary>
-        /// The Maximal Occurence number in the array.
+        /// The Maximal Occurence number in the array, null if the array
+        /// has an unbounded number of occurrences.
+        /// </summary>
+        public long? MaxOccur
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Shortcut to test if the array has no max occur.
+        /// </summary>
+        public bool IsUnbounded => !MaxOccur.HasValue;
+
+        /// <summary>
+        /// Path to the VariableSymbol of the DEPENDING ON clause if any.
         /// </summary>
-        public long MaxOccur
+        public string[] DependingOnPath
```

Comment:
Why not use SymbolReference ? This is the object we use everywhere to represents a reference to a Symbol.

Reply:
Hum... I am not sure that the depending on symbol references are, ArrayType's attributes, I don't agree, because type equality will not use this field, but this attibute is for the  VariableSymbol having an ArrayType and depending symbol references, so depending symbol references are VariableSymbol attributes.

Reply:
I agree that it's more a VariableSymbol attribute than an ArrayType attribute .
But see my comment on PR https://github.com/TypeCobolTeam/TypeCobol/pull/1722#issuecomment-631304559, we can remove all *Path attributes.

Reply:
See 99c458ed8962d3545b59ffe8c886a148a2d60a41.